### PR TITLE
Merge the core

### PR DESCRIPTION
**Pull request by houshuang**
I am basing this on mergeutils now to see the right diff, but mergeutils should be merged first, and then this should be merged into develop. It's a cleaned up version of the base unil changes, excluding activities/operators.

There are some things I still want to revise, like the way file uploading works, but I think this code is pretty much all good, and ready to merge. We need to make sure everything works well on develop etc, and the best way is to start working off develop as soon as possible. This also means that new features (logging etc) can be based directly off develop (and merged into unil if we need to keep a separate branch).

### CODE REVIEW

File: frog/client/main.html [hunk 1]
```diff
@@ -1,5 +1,9 @@
 <head>
   <title>FROG</title>
+  <script src="https://cdn.ravenjs.com/3.17.0/raven.min.js" crossorigin="anonymous"></script>
```
Comment (houshuang) [inline frog/client/main.html]: Do you know of an easy way to make this configurable? It does not yet have access to Meteor.settings at this stage.

File: frog/imports/api/uploads.js [hunk 2]
```diff
@@ -1,9 +1 @@
-// @flow
-
-import { FS } from 'meteor/cfs:base-package';
-
-export const Uploads = new FS.Collection('uploads', {
-  stores: [new FS.Store.FileSystem('uploads')]
-});
-
-Uploads.allow({ insert: () => true });
+export const Uploads = { insert: () => {} };
```
Comment (houshuang) [inline frog/imports/api/uploads.js]: Putting this here for now to make Flow happy. Will rethink a bit how to do this to make sure it works both in development and production.

File: frog/imports/ui/App/index.js [hunk 3]
```diff
@@ -66,7 +74,8 @@ class FROGRouter extends Component {
     const hasLogin = query.login;
 
     if (this.state.mode !== 'loggingIn') {
-      if (process.env.NODE_ENV !== 'production') {
+      if (true) {
```
Comment (houshuang) [inline frog/imports/ui/App/index.js]: Eventually I want to make this more secure again

File: frog/imports/api/uploads.js (hunk 2 above)
Comment (lfaucon) [reply frog/imports/api/uploads.js]: ok

File: frog/server/main.js [hunk 4]
```diff
@@ -19,19 +20,19 @@ import { Sessions } from '../imports/api/sessions.js';
 import { Products } from '../imports/api/products.js';
 import { Objects } from '../imports/api/objects.js';
 
+if (process.env.NODE_ENV !== 'production') {
+  // eslint-disable-next-line global-require
+  require('longjohn');
+}
+
+process.setMaxListeners(100);
```
Comment (lfaucon) [inline frog/server/main.js]: Did this make any change for us?

File: frog/server/sharedbCache.js [hunk 5]
```diff
@@ -0,0 +1 @@
+export const Cache = {};
```
Comment (lfaucon) [inline frog/server/sharedbCache.js]: Love that file :) Should it have flow annotation?

**Comment by lfaucon** [comment]
@houshuang I made a small change on how we generate random slugs. You can give it a look, then merge it

**Pull request merged**